### PR TITLE
fix(options-menu): added support for groups and titles

### PR DESCRIPTION
**Pull request by mcoker**
fixes https://github.com/patternfly/patternfly-next/issues/2361

### CODE REVIEW

**Comment by mceledonia** [comment]
Very nice, LGTM!

File: src/patternfly/components/OptionsMenu/examples/OptionsMenu.md [hunk 1]
```diff
@@ -143,6 +176,7 @@ import './OptionsMenu.css'
 | `role` or `aria` | `pf-c-options-menu` |  accessibility notes. |
 | `disabled` | `.pf-c-options-menu__toggle`, `.pf-c-options-menu__toggle-button` | Disables the options menu toggle and toggle button and removes it from keyboard focus. |
 *Note:* The attribute `aria-selected="true"` should be set programmatically to the selected item(s).
+| `role="separator"` | `li.pf-c-options-menu__separator` | Indicates that the separator is a separator. |
```
Comment (mattnolting) [inline src/patternfly/components/OptionsMenu/examples/OptionsMenu.md]:
```suggestion
| `role="separator"` | `li.pf-c-options-menu__separator` | Indicates that the list item is a separator. |
```

File: src/patternfly/components/OptionsMenu/examples/OptionsMenu.md [hunk 2]
```diff
@@ -152,10 +186,12 @@ import './OptionsMenu.css'
 | `.pf-c-options-menu__toggle-text` | `<span>` | Initiates a wrapper for toggle text.
 | `.pf-c-options-menu__toggle-icon` | `<i>` | Initiates the up/down arrow beside toggle text. |
 | `.pf-c-options-menu__toggle-button` | `<button>` | Initiates a custom toggle button for use when `.pf-c-options-menu__toggle` is a `<div>` or non-interactive element. |
-| `.pf-c-options-menu__menu` | `<ul>` |  Initiates the custom options-menu dropdown menu. |
-| `.pf-c-options-menu__menu-item` | `<li>` |  Initiates the items in the custom options-menu dropdown menu. |
+| `.pf-c-options-menu__menu` | `<ul>` |  Initiates the custom options-menu options-menu menu. |
```
Comment (mattnolting) [inline src/patternfly/components/OptionsMenu/examples/OptionsMenu.md]:
```suggestion
| `.pf-c-options-menu__menu` | `<ul>` |  Initiates the custom options-menu menu. |
```

File: src/patternfly/components/OptionsMenu/examples/OptionsMenu.md [hunk 3]
```diff
@@ -152,10 +186,12 @@ import './OptionsMenu.css'
...
+| `.pf-c-options-menu__menu-item` | `<li>` |  Initiates the items in the custom options-menu options-menu menu. |
```
Comment (mattnolting) [inline src/patternfly/components/OptionsMenu/examples/OptionsMenu.md]:
```suggestion
| `.pf-c-options-menu__menu-item` | `<li>` |  Initiates the items in the custom options-menu menu. |
```

**Comment by mcoker** [comment]
@mattnolting updated

**Comment by mcoker** [comment]
@mattnolting I think you can merge this one if you're OK with it.

**Pull request merged**